zipline/assets/asset_writer.py

```diff
@@ -348,28 +324,26 @@ def init_db(self, engine, constraints=True):
                 sa.Integer,
                 unique=True,
                 nullable=False,
-                primary_key=constraints,
+                primary_key=True,
             ),
             sa.Column('symbol', sa.Text),
             sa.Column(
                 'root_symbol',
                 sa.Text,
-                *((sa.ForeignKey(self.futures_root_symbols.c.root_symbol),)
-                  if constraints else ())
+                sa.ForeignKey(self.futures_root_symbols.c.root_symbol),
             ),
             sa.Column('asset_name', sa.Text),
-            sa.Column('start_date', sa.Integer, default=0),
-            sa.Column('end_date', sa.Integer),
-            sa.Column('first_traded', sa.Integer),
+            sa.Column('start_date', sa.Integer, default=0, nullable=False),
+            sa.Column('end_date', sa.Integer, nullable=False),
+            sa.Column('first_traded', sa.Integer, nullable=False),
             sa.Column(
                 'exchange',
                 sa.Text,
-                *((sa.ForeignKey(self.futures_exchanges.c.exchange),)
-                  if constraints else ())
+                sa.ForeignKey(self.futures_exchanges.c.exchange),
             ),
-            sa.Column('notice_date', sa.Integer),
-            sa.Column('expiration_date', sa.Integer),
-            sa.Column('auto_close_date', sa.Integer),
+            sa.Column('notice_date', sa.Integer, nullable=False),
+            sa.Column('expiration_date', sa.Integer, nullable=False),
+            sa.Column('auto_close_date', sa.Integer, nullable=False),
```

Comment:
Sorry I didn't explain this better in the PR itself - it's in [this commit message] (https://github.com/quantopian/zipline/commit/21366e8529f9df1f0348fd3bc43bd15e684a1085).  There's still a sentinel value for missing data, but now that we're using [vectorized operations] (https://github.com/quantopian/zipline/commit/21366e8529f9df1f0348fd3bc43bd15e684a1085#diff-33cdc2260291626609681cc61297818aR495) to serialize the dts, we're storing only ints, so the sentinel is NaT instead of NULL.